api/src/main/java/com/redhat/cloud/custompolicies/api/model/condition/expression/ExprParser.java

```diff
@@ -0,0 +1,377 @@
+                // What if we're talking about a number..?
+                if(ctx.value() != null) {
+                    // String contains
+                    if(ctx.string_compare_operator().CONTAINS() != null) {
+                        // Repetitive code, refactor at some point when more array operators are known
+                        if(targetValue instanceof Iterable) {
+                            return arrayContains((Iterable) targetValue, strValue);
+                        } else {
+                            return targetValueStr.contains(strValue);
+                        }
+                    }
+                }
+                if(ctx.array() != null) {
+                    boolean validForAll = true;
+                    for (ExpressionParser.ValueContext valueContext : ctx.array().value()) {
+                        String val = valueToString(valueContext);
+                        validForAll &= targetValueStr.contains(val);
+                    }
+                    return validForAll;
+                }
+                return false;
+            }
+
+            if(ctx.array_operator() != null) {
+                if(ctx.array() != null) {
+                    // We have restricted array values to be strings always
+                    if(ctx.string_compare_operator() != null) {
+                        if(ctx.string_compare_operator().CONTAINS() != null) {
+                            boolean validForAll = true;
+                            for (ExpressionParser.ValueContext valueContext : ctx.array().value()) {
+                                String val = valueToString(valueContext);
+                                if (targetValue instanceof Iterable) {
+                                    validForAll &= arrayContains((Iterable) targetValue, val);
+                                } else {
+                                    validForAll &= targetValueStr.contains(val);
+                                }
+                            }
+
+                            return validForAll;
+                        }
+                    }
+                    else if(ctx.array_operator() != null) {
+                        if(ctx.array_operator().IN() != null) {
+                            boolean validForAny = false;
+                            for (ExpressionParser.ValueContext valueContext : ctx.array().value()) {
+                                String val = valueToString(valueContext);
+                                validForAny |= targetValueStr.equals(val);
+                            }
+                            return validForAny;
+                        }
+                    }
+                }
+            }
+
+            // The define only check (negative is thrown out earlier)
+            return true;
+        }
+
+        private Object decodeKeyToValue(String eventField) {
+            Object sEventValue = null;
+            if (TENANT_ID.equals(eventField)) {
```

Comment:
Do we really need this, as we should only ever work on "our own" items. So if an event for tenant 123 is coming in, we should only ever apply rules for tenant 123

Reply:
Not really no if we never have any multi-tenant queries (where the tenant could include some context).

The default matching checks the tenantId in the Event matching (Drools rule).